src/components/Swap/Toolbar/Caption.tsx

```diff
@@ -139,10 +137,35 @@ export interface TradeProps {
 
 interface ExpandProps {
   expanded: boolean
-}
+  warning?: 'warning' | 'error'
+}
+
+const ExpanderRow = styled(Row)<{ $expanded: boolean; warning?: 'warning' | 'error' }>`
+  ${({ warning, $expanded }) =>
+    warning &&
```

Comment:
This is hard to follow because it's using defaulting - could you use an if-statement instead?

```
  ${({ warning, $expanded }) => {
    if (!warning) return undefined
    return css`
```

or

```
  if (warning) {
    return css`
      ...
    `
  }